# MDRD


Ci dessous je crée une fonction qui me renvoie le MDRD (une formule permettant d'estimer la fonction rénale d'un individu). Cette formule dépend de la créatinine, du sexe, de l'âge. 
Quand on dit que le patient a 120 µmole/litre de créatinine, cela veut dire qu'il à entre 119,51 et 120.50, soit 120 +/- 0.5. 
Quand on dit que le patient a 60 ans, cela veux dire qu'il a entre 60 et 61 ans. soit 60 +.5 +/- .5 
Je veux voir l'impact de ces petites fluctuations sur la sortie.

Dans cette page, j'ai crée une fonction mdrd qui fait le calcul et une fonction multi_mdrd, qui fait plusieurs tests et renvoie le min et le max. 

Note : dans cette page je ne traite pas des erreurs liées à l'analyse elle même.

In [45]:
from  math import pow 
import doctest

In [46]:
default_meth = 'non IDMS' # other : 'IDMS'

In [47]:
def mdrd(creat, age, sexe='M', meth=default_meth):
    """return MDRD result
    
    >>> round(mdrd(117, 49.1), 4)
    61.1565
    """
    if meth == 'IDMS':
        coef = 175
    else:
        coef = 186.3
    result = 0
    def common():
        return coef * pow(creat / 88.4, -1.154) * pow(age, -0.203)
    if sexe == 'M':
        result = common()
                    
    elif sexe == 'F':
        result = common() *  0.742
        pass
    else:
        return None
    return result

doctest.testmod()

TestResults(failed=0, attempted=1)

## Quelques exemples d'utilisation de la fonction mdrd.

In [48]:
mdrd(117, 49.1, sexe = 'F')

45.37813540082289

In [49]:
mdrd(117, 49.1, sexe = 'M', meth = '')

61.156516712699315

## Une fonction pour tester plusieurs valeurs à la fois 

In [50]:
def multi_mdrd(creat, age, sexe, meth=default_meth):
    """Calcul une sorte d'interval de confiance en tenant compte des arrondis possibles de données entrées.
    
Quand on importe des données, elles peuvent être arrondies. Par exemple, une créatine à 117 peut 
provenir d'une valeur initiale à 116.8 ou à 117.5.
    """
    lst = [mdrd(creat, age, sexe=sexe, meth=meth),
           mdrd(creat+0.5, age, sexe=sexe, meth=meth),
           mdrd(creat-0.5, age, sexe=sexe, meth=meth),
           mdrd(creat+0.5, age+1, sexe=sexe, meth=meth),
           mdrd(creat-0.5, age+1, sexe=sexe, meth=meth)           
          ]
    print(lst)
    mini = min(lst)
    maxi = max(lst)
    print ("min - max : {:.2f} - {:.2f}".format(mini, maxi))    
    

In [51]:
multi_mdrd(50,80, sexe = 'F')

[109.6185420681512, 108.367026359238, 110.89730880873964, 108.09409375563303, 110.61800344950966]
min - max : 108.09 - 110.90


Ci dessus, on voit que selon le type d'affichage et d'arrondi des données entrées, pour une valeur de créatinine de 50, un âge de 80 chez une femme, le résultat varie entre 108.09 et 110.90.


## Création d'une fonction plus générique  ?

Cela revient à définir un nombre non pas uniquement par sa valeur exacte mais par sa valeur et une incertitude absolue.
Il y a certainement une librairie qui traite de cela. 